scala/extends.go

```diff
@@ -0,0 +1,36 @@
+package scala
+
+import "strings"
+
+type ExtendsDeclaration struct {
+	className string
+	params    []string
```

Comment:
@eugene-yao-zocdoc consider `params` to be `[]Writable`, check method params for reference